File: solutions_5658571765186560_1/C++/wishlin/p4.cpp
#include <iostream>
#include <stdio.h>
using namespace std;

int main() {
  int cas,x,r,c,res;

  cin>>cas;
  for (int k=1; k<=cas; ++k) {
    cin>>x>>r>>c;
    res=0;

    if (x==1) {
      res = 1;
    }
    else if (x==2) {
      res = ((r%2==0)||(c%2==0));
    }
    else if (x==3) {
      res = ((r%3==0)||(c%3==0)) && (r>=2) && (c>=2);
    }
    else if (x==4) {
      res = ((r*c)%4==0) && (r>=3) && (c>=3);
    }
    else if (x<=6) {
      res = ((r*c)%x==0) && (r>=x-1) && (c>=x-1);
    }

    if (res) printf("Case #%d: GABRIEL\n", k);
    else printf("Case #%d: RICHARD\n", k);
  }
  return 0;
}

File: solutions_5658571765186560_0/java/qwp/P4.java
import java.util.*;

public class P4 {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int num = Integer.parseInt(in.nextLine());
		for (int i = 1; i <= num; i++) {
			String[] ss = in.nextLine().split(" ");
			int x = Integer.parseInt(ss[0]);
			int r = Integer.parseInt(ss[1]);
			int c = Integer.parseInt(ss[2]);
			
			String res = solve(x, r, c);
			System.out.println("Case #" + i + ": " + res);
		}
		in.close();
	}
	
	public static String solve(int x, int r, int c) {
		int s = Math.min(r,  c);
		int l = Math.max(r,  c);
		
		if ((s * l) % x != 0)
			return "RICHARD";
		if (l < x)
			return "RICHARD";
		if (x >= 8)
			return "RICHARD";
		if (s < (x + 1) / 2)
			return "RICHARD";
		if (x >= 4 && s <= (x + 1) / 2)
			return "RICHARD";
		
		return "GABRIEL";
	}
}
